aws/resource_aws_vpc_peering_connection.go

```diff
@@ -310,38 +304,30 @@ func resourceAwsVPCPeeringDelete(d *schema.ResourceData, meta interface{}) error
 	return err
 }
 
-// resourceAwsVPCPeeringConnectionStateRefreshFunc returns a resource.StateRefreshFunc that is used to watch
-// a VPCPeeringConnection.
-func resourceAwsVPCPeeringConnectionStateRefreshFunc(conn *ec2.EC2, id string) resource.StateRefreshFunc {
+func vpcPeeringConnectionRefreshState(conn *ec2.EC2, id string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
 		resp, err := conn.DescribeVpcPeeringConnections(&ec2.DescribeVpcPeeringConnectionsInput{
-			VpcPeeringConnectionIds: []*string{aws.String(id)},
+			VpcPeeringConnectionIds: aws.StringSlice([]string{id}),
 		})
 		if err != nil {
-			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidVpcPeeringConnectionID.NotFound" {
-				resp = nil
-			} else {
-				log.Printf("Error reading VPC Peering Connection details: %s", err)
-				return nil, "error", err
+			if isNoSuchVpcPeeringConnectionErr(err) {
+				return nil, ec2.VpcPeeringConnectionStateReasonCodeDeleted, nil
 			}
-		}
 
-		if resp == nil {
-			// Sometimes AWS just has consistency issues and doesn't see
-			// our instance yet. Return an empty state.
-			return nil, "", nil
+			return nil, "", err
 		}
 
 		pc := resp.VpcPeeringConnections[0]
```

Comment:
We should have `nil`/`len()` checks for `resp`, `resp.VpcPeeringConnections`, and `pc.Status` to prevent panics.

Reply:
Will do.